File: app/data_sources/ReportDataSource.scala
package data_sources

import net.sf.jasperreports.engine.JRDataSource

/**
 * Interface of all the data sources. Used by the report generators.
 * @author Jorge Migueis
 */
trait ReportDataSource {

  def convertToJRDataSource():JRDataSource

  def reportMatcher():String

}

File: src/main/java/dataSources/ReportDataSource.java
package dataSources;

import net.sf.jasperreports.engine.JRDataSource;

/**
 * Created by peterwhitehead on 04/05/2016.
 */
public interface ReportDataSource {
    JRDataSource convertToJRDataSource();

    String jasperReportFilenameMatcher();
}
